bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,271 @@
+            r"(?P<open>{{)(?P<content>.*?)(?P<close>}})")
+        control_start_regex = re.compile(
+            r"(?P<open>{%)(?P<content>\s*(for|if).*?)(?P<close>%})")
+        control_end_regex = re.compile(
+            r"(?P<open>{%)(?P<content>\s*end(for|if)\s*?)"
+            r"(?P<close>%})(?P<label>{#.*?#})?")
+
+        # Whenever a control construct starts a tuple of the
+        # expected end label and line number are put on this stack.
+        # Whenever an end tag is encountert the item on top is popped.
+        control_stack = []
+
+        for line_number, line in enumerate(file, start=1):
+
+            # check for variable spacing issues
+            for m in variable_regex.finditer(line):
+                match = m.group("content")
+                if not is_spacing_valid(match, variable_spacing):
+                    diff = generate_spacing_diff(
+                        file, filename, line, line_number, m, variable_spacing)
+                    yield Result.from_values(
+                        origin=self,
+                        message="Variable blocks should be spaced with "
+                                "`{0}` spaces on each side.".format(
+                                    variable_spacing),
+                        file=filename,
+                        line=line_number,
+                        column=m.start(0) + 1,
+                        end_line=line_number + 1,
+                        end_column=m.end(0) + 1,
+                        diffs=diff)
+
+            # check for control start tags
+            for m in control_start_regex.finditer(line):
+                # build the label which is expected at the end of this block
+                end_label = "{{#{content}#}}".format(
+                    content=m.group("content"))
+                control_stack.append((end_label, line_number))
+
+                # check for spacing issues with this tag
+                if not is_spacing_valid(m.group("content"), control_spacing):
+                    yield self.handle_control_spacing_issue(
+                        file, filename, line, line_number, control_spacing, m)
+
+            # check for control end tags
+            for m in control_end_regex.finditer(line):
+                label = m.group("label")
+                try:
+                    (expected_label, start_in_line) = control_stack.pop()
+                except IndexError:
+                    # and end tag without an opening tag
+                    yield Result.from_values(
+                        origin=self,
+                        message="Control end tag has no corresponding start",
+                        file=filename,
+                        line=line_number,
+                        column=m.start(0) + 1,
+                        end_line=line_number + 1,
+                        end_column=m.end(0) + 1,
+                        severity=RESULT_SEVERITY.MAJOR)
+                    continue
```

Comment:
Can you put a comment here saying "Continue to next line as label checking can give false results" because if someone wants to add another feature here later, they need to be aware to put it on top